### PR TITLE
Add Delete Plan feature

### PR DESCRIPTION
**Pull request by TianaiYan**

### CODE REVIEW

File: src/main/java/werkit/UI.java [hunk 1]
```diff
@@ -429,6 +430,18 @@ public void printNewPlanCreatedMessage(Plan newPlan) {
         System.out.println();
     }
 
+    /**
+     * Prints a message when a plan has been deleted and shows the deleted workout.
+     *
+     * @param deletedPlan The plan to be deleted.
+     */
+    public void printDeletePlanMessage(Plan deletedPlan) {
```
Comment (a1021492980) [inline src/main/java/werkit/UI.java:438]: Good abstraction!

**Review by a1021492980 — COMMENTED** [review]
No issue from me, good implementation.

File: src/main/java/data/plans/PlanList.java [hunk 2]
```diff
@@ -174,4 +176,36 @@ public void listAllPlan() {
...
+
+        if (!isIndexToDeleteValid) {
+            throw new InvalidPlanException(className, InvalidPlanException.PLAN_INDEX_OUT_OF_RANGE);
+        }
+
+        assert (indexToDelete > 0) && (indexToDelete <= plansDisplayList.size());
+        Plan deletedPlan = getPlanFromIndexNum(indexToDelete);
+        String deletedPlanKey = deletedPlan.toString();
+        plansDisplayList.remove(indexToDelete - 1);
+        getPlansDisplayList().remove(deletedPlanKey);
```
Comment (Musfirahe0556596) [inline src/main/java/data/plans/PlanList.java:204]: is this supposed to be getPlansHashMapList() as I notice only the plan gets removed from the plans list (display list), not the hashmap?

**Review by Musfirahe0556596 — APPROVED** [review]
Just a consideration! But everything looks great!

File: src/main/java/data/plans/PlanList.java (hunk 2 above)
Comment (TianaiYan) [reply src/main/java/data/plans/PlanList.java:204]: Yes, it should be getPlansHashMapList(). I didn't notice that. Thank you for pointing it out.

**Pull request merged**